drivers/Ticker.h

```diff
@@ -75,10 +75,15 @@ class Ticker : public TimerEvent, private NonCopyable<Ticker> {
 
     /** Attach a function to be called by the Ticker, specifying the interval in seconds
      *
+     *  The method forwards its arguments to attach_us() rather than copying them which
+     *  may not be trivial depending on the callback copied.
+     *  The function is forcibly inlined to not use floating-point operations. This is
+     *  possible given attach_us() expects an integer value for the callback interval.
      *  @param func pointer to the function to be called
      *  @param t the time between calls in seconds
      */
-    void attach(Callback<void()> func, float t)
+    template <typename F, typename T>
+    MBED_FORCEINLINE void attach(F&& func, T&& t)
```

Comment:
Hmm, hadn't occurred to me to templatise the actual time. I can't immediately see a particular benefit to that. Actually, it will get in the way of future `std::duration` overloads, so it's not acceptable.

If you really wanted it, you'd have to qualify it with an `enable_if<is_arithmetic>` or something, but I think templatising that parameter is just complication for no reason.

Any reason to omit the `std::forward<F>`? Even if it doesn't make a difference now, it should make a difference with some upcoming `Callback` optimisations.

Reply:
I un-templatized the time interval.
See the size difference below.
Thanks

With reference forwarding

```
| Module             |      .text |    .data |      .bss |
|--------------------|------------|----------|-----------|
| [fill]             |    157(+0) |    2(+0) |  2187(+0) |
| [lib]/c.a          |   6892(+0) | 2108(+0) |    89(+0) |
| [lib]/gcc.a        |    772(+0) |    0(+0) |     0(+0) |
| [lib]/misc         |    180(+0) |    4(+0) |    28(+0) |
| main.o             |  13686(+0) |    0(+0) |  1337(+0) |
| mbed-os/components |     34(+0) |    0(+0) |     0(+0) |
| mbed-os/drivers    |    856(+0) |    0(+0) |     0(+0) |
| mbed-os/features   | 109465(+0) |    8(+0) |    12(+0) |
| mbed-os/hal        |   1560(+0) |    4(+0) |    67(+0) |
| mbed-os/platform   |   4877(+0) |  262(+0) |   478(+0) |
| mbed-os/rtos       |   6570(+0) |  168(+0) |  5973(+0) |
| mbed-os/targets    |   5586(+0) |   12(+0) |   381(+0) |
| Subtotals          | 150635(+0) | 2568(+0) | 10552(+0) |
Total Static RAM memory (data + bss): 13120(+0) bytes
Total Flash memory (text + data): 153203(+0) bytes
```


Without reference forwarding
```
| Module             |        .text |    .data |      .bss |
|--------------------|--------------|----------|-----------|
| [fill]             |      161(+4) |    2(+0) |  2187(+0) |
| [lib]/c.a          |     6892(+0) | 2108(+0) |    89(+0) |
| [lib]/gcc.a        |      772(+0) |    0(+0) |     0(+0) |
| [lib]/misc         |      180(+0) |    4(+0) |    28(+0) |
| main.o             |  14100(+414) |    0(+0) |  1337(+0) |
| mbed-os/components |       0(-34) |    0(+0) |     0(+0) |
| mbed-os/drivers    |      856(+0) |    0(+0) |     0(+0) |
| mbed-os/features   |   109465(+0) |    8(+0) |    12(+0) |
| mbed-os/hal        |     1560(+0) |    4(+0) |    67(+0) |
| mbed-os/platform   |     4877(+0) |  262(+0) |   478(+0) |
| mbed-os/rtos       |     6570(+0) |  168(+0) |  5973(+0) |
| mbed-os/targets    |     5586(+0) |   12(+0) |   381(+0) |
| Subtotals          | 151019(+384) | 2568(+0) | 10552(+0) |
Total Static RAM memory (data + bss): 13120(+0) bytes
Total Flash memory (text + data): 153587(+384) bytes
```